src/api/routes/inscriptions.ts

```diff
@@ -0,0 +1,265 @@
+import { TypeBoxTypeProvider } from '@fastify/type-provider-typebox';
+import { Type } from '@sinclair/typebox';
+import { Value } from '@sinclair/typebox/value';
+import { FastifyPluginAsync, FastifyPluginCallback } from 'fastify';
+import { Server } from 'http';
+import {
+  InscriptionResponse,
+  LimitParam,
+  NotFoundResponse,
+  OffsetParam,
+  PaginatedResponse,
+  MimeTypesParam,
+  SatoshiRaritiesParam,
+  OutputParam,
+  OrderByParam,
+  OrderParam,
+  OrderBy,
+  Order,
+  InscriptionIdentifierParam,
+  BlockHashParamCType,
+  BlockHeightParamCType,
+  InscriptionIdParamCType,
+  BlockHeightParam,
+  BlockParam,
+  OrdinalParam,
+  InscriptionNumberParam,
+  TimestampParam,
+  AddressesParam,
+  InscriptionIdsParam,
+  InscriptionNumbersParam,
+  InscriptionLocationResponse,
+} from '../schemas';
+import { handleInscriptionCache, handleInscriptionTransfersCache } from '../util/cache';
+import {
+  DEFAULT_API_LIMIT,
+  hexToBuffer,
+  parseDbInscription,
+  parseDbInscriptions,
+  parseInscriptionLocations,
+} from '../util/helpers';
+
+function blockParam(param: string | undefined, name: string) {
+  const out: { [k: string]: string } = {};
```

Comment:
nit: Record types are bit more idiomatic typescript
```suggestion
  const out: Record<string, string> = {};
```